catalyst-gateway/bin/src/logger.rs

```diff
@@ -33,14 +33,13 @@ impl From<LogLevel> for tracing::Level {
     }
 }
 
-impl LogLevel {
-    /// Map [`LogLevel`] to [`tracing::Level`]
-    pub(crate) fn as_log_level(self) -> tracing::log::LevelFilter {
-        match self {
-            LogLevel::Info => tracing::log::LevelFilter::Info,
-            LogLevel::Debug => tracing::log::LevelFilter::Debug,
-            LogLevel::Warn => tracing::log::LevelFilter::Warn,
-            LogLevel::Error => tracing::log::LevelFilter::Error,
+impl From<LogLevel> for tracing::log::LevelFilter {
+    fn from(val: LogLevel) -> Self {
+        match val {
+            LogLevel::Debug => Self::Debug,
+            LogLevel::Info => Self::Info,
+            LogLevel::Warn => Self::Warn,
+            LogLevel::Error => Self::Error,
```

Comment:
Instead of a 1:1 mapping, could we not just use `tracing::log::LevelFilter` as the options type?  Why do we need our own type ?

Reply:
During the logger init process tracing uses two different LevelFilters, one for subscriber ` subscriber.with_max_level(...)` which is defined in the tracing library itself, another one for global logging `tracing::log::set_max_level(log_level.into());` which actually defined in another library `log`, and these two types are incompatible and don't have convenient From, Into trait implementation. So I found this is the easiest way how to deal with them